charts/replicated-library/templates/lib/_container.tpl

```diff
@@ -64,25 +64,37 @@
   resources:
     {{- toYaml . | nindent 4 }}
   {{- end }}
-{{- if $containerValues.probes -}}
-  {{- if $containerValues.probes.livenessProbe -}}
-  {{- with (mergeOverwrite $.Values.defaults.probes.livenessProbe $containerValues.probes.livenessProbe) }}
+  {{- if $containerValues.livenessProbe -}}
+  {{- with (mergeOverwrite $.Values.defaults.probes.livenessProbe $containerValues.livenessProbe) }}
   livenessProbe:
     {{- toYaml . | nindent 4 }}
   {{- end }}
+  {{- else if hasKey $containerValues "livenessProbe" -}}
+  {{- else if (first $containerValues.ports).containerPort -}}
```

Comment:
Curious if someone omits `ports` from their spec will you get a nil pointer error here?

Reply:
Good question I'll check.

Reply:
Yup, you were right Nil pointer. Just pushed a fix.